Source/WebCore/page/ContextMenuController.cpp

```diff
@@ -965,6 +973,11 @@ void ContextMenuController::populate()
         if (selectionIsInsideImageOverlay || (linkURL.isEmpty() && mediaURL.isEmpty() && imageURL.isEmpty())) {
             if (!imageURL.isEmpty())
                 appendItem(*separatorItem(), m_contextMenu.get());
+            
+            auto* ownerElement = frame->ownerElement();
+            bool isPDFDocument = ownerElement && ownerElement->document().isPDFDocument();
+            bool isMainFrame = frame->isMainFrame();
+            auto* page = frame->page();
```

Comment:
Let's use `RefPtr` instead of `auto*`, and move this variable up next to `ownerElement`.

Reply:
```
RefPtr page = frame->page();
RefPtr ownerElement = frame->ownerElement();
bool isPDFDocument = ownerElement && ownerElement->document().isPDFDocument();
bool isMainFrame = frame->isMainFrame();
```